crates/ruff_dev/src/generate_options.rs

```diff
@@ -32,56 +36,69 @@ fn generate_set(output: &mut String, set: &Set) {
 
     // Generate the fields.
     for (name, field) in &fields {
-        emit_field(output, name, field, set.name());
+        emit_field(output, name, field, set);
         output.push_str("---\n\n");
     }
 
     // Generate all the sub-sets.
     for (set_name, sub_set) in &sets {
-        generate_set(output, &Set::Named(set_name, *sub_set));
+        generate_set(output, &Set::Named(set_name, *sub_set, set.level() + 1));
     }
 }
 
 enum Set<'a> {
     Toplevel(OptionSet),
-    Named(&'a str, OptionSet),
+    Named(&'a str, OptionSet, u32),
 }
 
 impl<'a> Set<'a> {
     fn name(&self) -> Option<&'a str> {
         match self {
             Set::Toplevel(_) => None,
-            Set::Named(name, _) => Some(name),
+            Set::Named(name, _, _) => Some(name),
         }
     }
 
     fn title(&self) -> &'a str {
         match self {
             Set::Toplevel(_) => "Top-level",
-            Set::Named(name, _) => name,
+            Set::Named(name, _, _) => name,
         }
     }
 
     fn metadata(&self) -> &OptionSet {
         match self {
             Set::Toplevel(set) => set,
-            Set::Named(_, set) => set,
+            Set::Named(_, set, _) => set,
+        }
+    }
+
+    fn level(&self) -> u32 {
+        match self {
+            Set::Toplevel(_) => 0,
+            Set::Named(_, _, level) => *level,
         }
     }
 }
 
-fn emit_field(output: &mut String, name: &str, field: &OptionField, group_name: Option<&str>) {
-    // if there's a group name, we need to add it to the anchor
-    if let Some(group_name) = group_name {
+fn emit_field(output: &mut String, name: &str, field: &OptionField, parent_set: &Set) {
+    let header_level = if parent_set.level() < 2 {
```

Comment:
Add proper sub-headings for options under `lint`. We actually don't need this yet except if we decide to show the `lint.*` options on the website.